pkg/ccl/backupccl/generative_split_and_scatter_processor.go

```diff
@@ -72,24 +79,33 @@ func newGenerativeSplitAndScatterProcessor(
 	post *execinfrapb.PostProcessSpec,
 	output execinfra.RowReceiver,
 ) (execinfra.Processor, error) {
-
 	db := flowCtx.Cfg.DB
-	kr, err := MakeKeyRewriterFromRekeys(flowCtx.Codec(), spec.TableRekeys, spec.TenantRekeys,
-		false /* restoreTenantFromStream */)
-	if err != nil {
-		return nil, err
-	}
+	numSplitWorkers := int(spec.NumNodes)
+	numSecondSplitWorkers := 2 * numSplitWorkers
+
+	var scatterers []splitAndScatterer
+	for i := 0; i < numSplitWorkers+numSecondSplitWorkers; i++ {
+		var scatterer splitAndScatterer
+		if spec.ValidateOnly {
+			nodeID, _ := flowCtx.NodeID.OptionalNodeID()
+			scatterer = noopSplitAndScatterer{nodeID}
+		} else {
+			kr, err := MakeKeyRewriterFromRekeys(flowCtx.Codec(), spec.TableRekeys, spec.TenantRekeys,
+				false /* restoreTenantFromStream */)
+			if err != nil {
+				return nil, err
+			}
+			scatterer = makeSplitAndScatterer(db.KV(), kr)
+		}
 
-	scatterer := makeSplitAndScatterer(db.KV(), kr)
-	if spec.ValidateOnly {
-		nodeID, _ := flowCtx.NodeID.OptionalNodeID()
-		scatterer = noopSplitAndScatterer{nodeID}
+		scatterers = append(scatterers, scatterer)
 	}
 	ssp := &generativeSplitAndScatterProcessor{
-		flowCtx:   flowCtx,
-		spec:      spec,
-		output:    output,
-		scatterer: scatterer,
+		flowCtx:               flowCtx,
+		spec:                  spec,
+		output:                output,
+		scatterers:            scatterers[:numSplitWorkers],
```

Comment:
sanity checking that this creates a copy of the scatterers slice and doesn't just copy the pointer

Reply:
these do just copy the pointer but I changed the code to use two slices just so that we are not confused in the future.